_resources/port1.0/group/gnuradio-1.0.tcl

```diff
@@ -0,0 +1,201 @@
+
+if {[string first "37" $subport] > 1} {
+    default gnuradio.python_versions { 2.7 }
+    default gnuradio.default_python_variant +python27
+} else {
+    # the following versions use all the same python
+    default gnuradio.python_versions { 3.8 3.9 3.10 }
+    default gnuradio.default_python_variant +python39
+}
+
+# use C/C++11
+compiler.c_standard   2011
+compiler.cxx_standard 2011
+boost.version 1.71
+
+# see https://github.com/macports/macports-ports/pull/7805
+license_noconflict-append openssl
+
+# see https://github.com/macports/macports-ports/pull/7805
+license_noconflict-append graphviz
+
+# Define the available variants
+foreach py_ver ${gnuradio.python_versions} {
+    set py_ver_no_dot [join [split ${py_ver} "."] ""]
+    set variant_line {variant python${py_ver_no_dot} description "Build with python ${py_ver} support"}
+    foreach py_over ${gnuradio.python_versions} {
+        if { ${py_ver} == ${py_over} } { continue }
+        set py_over_no_dot [join [split ${py_over} "."] ""]
+        append variant_line " conflicts python${py_over_no_dot}"
+    }
+    append variant_line { { } }
+    eval $variant_line
+    if {[variant_isset python${py_ver_no_dot}]} {
+        if {${gnuradio.default_python_variant} != "+python${py_ver_no_dot}"} {
+            set gnuradio.default_python_variant ""
+        }
+    }
+}
+
+# set default python variant if not selected
+if {${gnuradio.default_python_variant} != ""} {
+    default_variants-append "${gnuradio.default_python_variant}"
+}
+
+# If a python variant is enabled, activate it
+set active_python_version ""
+set active_python_version_no_dot ""
+foreach py_ver ${gnuradio.python_versions} {
+    set py_ver_no_dot [join [split ${py_ver} "."] ""]
+    if {[variant_isset python${py_ver_no_dot}]} {
+        set active_python_version        ${py_ver}
+        set active_python_version_no_dot ${py_ver_no_dot}
+    }
+}
+
+depends_build-append \
+    port:pkgconfig
+
+depends_lib-append \
+    port:fftw-3-single \
+    path:lib/libmpir.dylib:mpir \
+    path:lib/libvolk.dylib:volk \
+    port:gmp \
+    port:python${active_python_version_no_dot}
+
+# need matplotlib for polar encoder/decoder, runtime only
+depends_run-append \
+    port:py${active_python_version_no_dot}-matplotlib \
+    port:py${active_python_version_no_dot}-opengl \
+    port:py${active_python_version_no_dot}-scipy \
+    port:py${active_python_version_no_dot}-numpy \
+    port:py${active_python_version_no_dot}-cheetah
+
+if {[string first "37" $subport] > 1} {
+    # still require cppunit for testing; NOTE: cppunit is checked for
+    # during configure, so we need it to be in depends_lib or
+    # depends_build to be used correctly. Choose the latter since it's
+    # not required for runtime; just for build/test.
+    depends_build-append \
+        port:cppunit
+} else {
+    # gr-vocoder removed in-tree libgsm, libcodec2, use system-wide libs
+    depends_lib-append \
+        port:codec2 \
```

Comment:
I think these dependencies (all in this `else` clause) belong in the GR Portfile, not here, since they are not common to all GR OOT modules.

Reply:
right. moved back. thank you